File: app/src/main/java/ys/moire/infra/prefs/PrefsImpl.java
package ys.moire.infra.prefs;

import android.app.Activity;
import android.content.Context;
import android.content.SharedPreferences;

import ys.moire.common.config.Config;

/**
 * Prefs.
 */

public class PrefsImpl implements Prefs {

    private final SharedPreferences preferences;

    public PrefsImpl(final Context context) {
        preferences = context
                .getApplicationContext()
                .getSharedPreferences(Config.PREF, Activity.MODE_PRIVATE);
    }

    @Override
    public void put(final String key, final String value) {
        preferences.edit().putString(key, value).apply();
    }

    @Override
    public void put(final String key, final int value) {
        preferences.edit().putInt(key, value).apply();
    }

    @Override
    public String get(final String key, final String defValue) {
        return preferences.getString(key, defValue);
    }

    @Override
    public int get(final String key, final int defValue) {
        return preferences.getInt(key, defValue);
    }
}

File: app/src/main/java/ys/moire/infra/prefs/PrefsImpl.kt
package ys.moire.infra.prefs

import android.app.Activity
import android.content.Context
import android.content.SharedPreferences

/**
 * Prefs.
 */

class PrefsImpl(context: Context) : Prefs {

    private val preferences: SharedPreferences = context
            .applicationContext
            .getSharedPreferences(ys.moire.common.config.PREF(), Activity.MODE_PRIVATE)

    override fun put(key: String, value: String) {
        preferences.edit().putString(key, value).apply()
    }

    override fun put(key: String, value: Int) {
        preferences.edit().putInt(key, value).apply()
    }

    override fun get(key: String, defValue: String): String = preferences.getString(key, defValue)

    override fun get(key: String, defValue: Int): Int = preferences.getInt(key, defValue)
}
